plugins/org.wso2.developerstudio.visualdatamapper.diagram/src/org/wso2/developerstudio/datamapper/diagram/xslt/xmltree/OutPutNode.java

```diff
@@ -0,0 +1,109 @@
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.developerstudio.datamapper.diagram.xslt.xmltree;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+import org.w3c.dom.Node;
+
+import static org.wso2.developerstudio.datamapper.diagram.xslt.config.XSLTGeneratorConstants.ATTRIBUTES_INITIALS;
+import static org.wso2.developerstudio.datamapper.diagram.xslt.config.XSLTGeneratorConstants.PROPERTIES_LOWER_CASE;
+import static org.wso2.developerstudio.datamapper.diagram.xslt.config.XSLTGeneratorConstants.KEY;
+import static org.wso2.developerstudio.datamapper.diagram.xslt.config.XSLTGeneratorConstants.VALUE;
+import static org.wso2.developerstudio.datamapper.diagram.xslt.config.XSLTGeneratorConstants.NODE;
+import static org.wso2.developerstudio.datamapper.diagram.xslt.config.XSLTGeneratorConstants.IN_NODE;
+import static org.wso2.developerstudio.datamapper.diagram.xslt.config.XSLTGeneratorConstants.NAME;
+
+/**
+ * This class represent a node in the output tree
+ * 
+ */
+public class OutPutNode extends XMLNode {
+    private final String xPath;
+
+    public InNode getInNode() {
+        return inNode;
+    }
+
+    private InNode inNode;
+    private final ArrayList<OutPutNode> childNodes;
+    private final HashMap<String, String> properties;
+
+    public OutPutNode(Node node, String parentXPath) {
+        this.name = getAttribute(node, NAME);
+        this.properties = new HashMap<>();
+        this.childNodes = new ArrayList<>();
+        if (parentXPath.equals("")) {
+            this.xPath = parentXPath + this.name;
+        } else {
+            this.xPath = parentXPath + "/" + this.name;
+        }
+        populate(node);
+    }
+
+    public String getProperty(String property) {
+        if (this.properties.containsKey(property)) {
+            return this.properties.get(property);
+        }
+        return null;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public HashMap<String, String> getProperties() {
+        return properties;
+    }
+
+    /**
+     * This method extracts all the properties and build child nodes
+     * 
+     * @param node document node that represents the output node
+     */
+    private void populate(Node node) {
+        for (int i = 0; i < node.getChildNodes().getLength(); i++) {
+            Node childNode = node.getChildNodes().item(i);
+            if (childNode.getAttributes() != null) {
+                switch (childNode.getNodeName()) {
+                case PROPERTIES_LOWER_CASE:
```

Comment:
What operators are defined as "PROPERTIES_LOWER_CASE". As discussed on the review better use a meaningful name.